lib/player.js

```diff
@@ -2554,6 +2554,20 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       movePlayhead: (delta) => { mediaElement.currentTime += delta; },
     });
 
+    if (this.cmcdManager_) {
+      if (mediaElement.autoplay && mediaElement.muted) {
```

Comment:
In the browser you can configure that no video is played automatically in any situation, so this if is wrong in that case.

Reply:
To detect whether autoplay is effectively happening, I'm considering two options:
- Listen for click and keydown events to check if the user interacted with the video. If there was no user interaction, I would include the load time in msd. If the user did interact, I would not include the load time.
- Use the promise returned by mediaElement.play(). If autoplay is blocked, the promise will reject, and I won't include the load time in msd.

Neither option fully convinces me. Do you think one of these approaches would work, or do you have another suggestion?

Reply:
@joeyparrish @littlespex Any idea/suggestion?

Reply:
I think using the play() promise is more reasonable than watching user interactions.  Not sure exactly which call to play() you want to instrument, though.

I had thought there was a method to detect autoplay policy, but it seems that it never got standardized, and only Firefox has it.